Feat/update latest speakers

ShugKnight24 opened the pull request:
# Pull Request

## Description

- Adds the initial confirmed speakers for 2025 devfest
- Add prior year photos and optimize via imageOptim
- Remove space from Chris Demars photo name
- Correct spelling of 'deprecated' directory
- Rename all assets with underscores
- Optimize all assets with imageOptim
- Improve spacing of speakers cards on mobile
- Add hardcoded colors to chevron & twitterhandle to ensure dark mode doesn't cause issues
- Add 2023 sponsors
- Make layout sections headers inline to condense space and match -> https://gdg-summit-webiste.vercel.app/
- Render SessionCard data conditionally
  - Make all session cards the same width regardless of title length

## Type of Change

- [X] New feature (non-breaking change which adds functionality)
- [X] Code refactoring

## Related Issues

Closes #66 

## Code Quality Checklist

<!-- Mark completed items with an "x" -->

- [X] Code follows the project's coding standards
- [X] Self-review of the code has been performed
- [X] Code is properly commented where necessary
- [X] No console.log statements left in the code
- [X] No hardcoded values that should be configurable
- [X] Error handling is implemented where appropriate

## Testing

<!-- Mark completed items with an "x" -->

- [X] Changes have been tested locally
- [X] All existing tests pass (if applicable)
- [ ] New tests have been added for new functionality (if applicable)
- [X] Manual testing has been performed
- [X] Accessibility testing has been performed (for UI changes)

## Accessibility (for UI changes)

<!-- Mark completed items with an "x" -->

- [ ] All interactive elements are keyboard accessible
- [ ] Color contrast meets WCAG AA standards
- [ ] Images have appropriate alt text
- [ ] Form labels are properly associated
- [ ] Focus indicators are visible
- [ ] Screen reader compatibility has been tested

## Build & Deployment

<!-- Mark completed items with an "x" -->

- [X] Application builds successfully
- [X] No build warnings or errors
- [X] Dependencies are up to date
- [X] No security vulnerabilities introduced

## Screenshots (if applicable)

### Sessions

<img width="1230" height="360" alt="Screenshot 2025-10-09 at 7 26 09 PM" src="https://github.com/user-attachments/assets/38c0668c-1820-4e2d-b82b-0e0a492f476e" />
<img width="1203" height="338" alt="Screenshot 2025-10-09 at 7 35 11 PM" src="https://github.com/user-attachments/assets/e993c6f6-f48e-4428-a237-46e166726abe" />
<img width="1024" height="305" alt="Screenshot 2025-10-09 at 7 35 08 PM" src="https://github.com/user-attachments/assets/322a2383-2faf-40d5-915b-2dde534454da" />


### Speakers

<img width="1177" height="414" alt="Screenshot 2025-10-09 at 7 35 01 PM" src="https://github.com/user-attachments/assets/6908f8e1-5661-4176-abb3-99ad79612caa" />

### 2023 sponsors

<img width="1240" height="793" alt="Screenshot 2025-10-10 at 12 41 43 PM" src="https://github.com/user-attachments/assets/a3a198d5-3b6d-4a21-acc6-567fb91dece5" />

### Layout Headers

#### Previously

<img width="1382" height="177" alt="Screenshot 2025-10-10 at 12 39 31 PM" src="https://github.com/user-attachments/assets/6447ea3d-cd1f-4d75-83d2-506ca998274c" />

#### New

<img width="1401" height="133" alt="Screenshot 2025-10-10 at 12 40 02 PM" src="https://github.com/user-attachments/assets/6d041792-c124-41ec-bb58-2db370bdd9b9" />


## Additional Notes

<!-- Any additional information that reviewers should know -->

---

**Note**: This PR will be automatically checked for:

- Code quality (ESLint)
- Code formatting (Prettier)
- Build verification
- Accessibility compliance
- Security vulnerabilities

All checks must pass before this PR can be merged.

ShugKnight24 committed:
feat: add initial 2025 speakers

ShugKnight24 committed:
refactor: update `SessionCard` to handle unknown values

- Make aria-controls and aria-label dependant on dynamic data it looks 
for

ShugKnight24 committed:
refactor: add on file images for prior year speakers

- Optimize assets using imageOptim
- Rename ChrisDemars photo to remove whitespace

ShugKnight24 committed:
chore: correct spelling of `deprecated` directory

shrinkray commented:
For the Sessions Section ... in dark mode the text colors of the sessions is white on white

shrinkray commented on src/components/sessions/SessionCard.jsx:
@ShugKnight24 This is a possible way to improve the design using light/dark mode colors for this. Use colors that you think fit this below is one solution. 

Card Background (line 46): 
- dark:bg-gray-800 and explicit border colors border-gray-200 dark:border-gray-700 
- so the card has a dark background in dark mode

Session Title (line 78): 
- Added text-gray-900 dark:text-white for proper contrast in both modes

Speaker Names (line 81): 
- text-gray-700 dark:text-gray-300 for readability

"at" and "in" labels (lines 84, 90):  text-gray-900 dark:text-gray-100

Time Display (line 85): 
- Add dark:text-slate-400 to complement the existing text-slate-500

Room Name (line 91): 
- text-gray-900 dark:text-gray-100

Chevron Icon (line 100): 
- text-gray-900 dark:text-gray-100 so the icon is visible in both modes

Description Text (line 111): 
- text-gray-900 dark:text-gray-100 and dark:border-gray-600 for the border

Light mode: White background with dark text
Dark mode: Dark gray background (gray-800) with light text

ShugKnight24 replied:
@shrinkray I think your PR mitigated a lot of the issues. You hardcoded background colors which prevents the issues.

I hardcoded a couple of things like the session card chevron, the twitter handle so they'd work correctly in dark mode and matched the speakers chevron the prior update you made

Can make some adjustments if you'd prefer

shrinkray commented on src/assets/images/speakers/Chris_Demars.jpg:
Will images need to be resized by Tailwind? If so, please resize and where possible compress as you do for those images, so they are not wildly scaled.

shrinkray commented on src/data/2023/speakers.js:
Great catches on image names!

ShugKnight24 replied:
tried to catch them all now, might have missed some

shrinkray reviewed (commented):
Primary issue is on color. please test with dark mode.

ShugKnight24 commented:
Good points @shrinkray

Let's get your PR merged and see what needs to be adjusted from there

ShugKnight24 committed:
Merge branch '2025' into feat/update_latest_speakers

ShugKnight24 committed:
refactor: improve styling

- Hardcode colors to be consistent between light and dark mode
- Improve layout of speaker cards by adding margin to the botton and 
centering when mobile

ShugKnight24 committed:
chore: optimize all assets, replace spaces with underscore in name

- optimize all assets with imageOptim
- rename to use underscore, replace name where used

ShugKnight24 committed:
feat: add sponsors23 data to previous events

ShugKnight24 committed:
chore: remove unused image comment

ShugKnight24 committed:
refactor: make all layout headings inline

- Make consistent to other summit website

shrinkray commented:
@ShugKnight24 For the speaker data that is TBD, should we just not show that content now? I think like we can promote the speaker before we know when they are speaking. Could we either comment out that content or if it returns TBD, not show the line?

ShugKnight24 committed:
refactor: make SessionCard data display conditionally

- Improve display so all cards are the same width

ShugKnight24 commented:
> @ShugKnight24 For the speaker data that is TBD, should we just not show that content now? I think like we can promote the speaker before we know when they are speaking. Could we either comment out that content or if it returns TBD, not show the line?

Good call - rendering it conditionally now

shrinkray reviewed (dismissed):
Tested in browser, Viewed code in GitHub, LGTM

ShugKnight24 committed:
chore: add Michael Bower headshot

shrinkray reviewed (approved):
LGTM
